### PR TITLE
New Quest Reward Flow

### LINKED ISSUE #573: New Quest Reward Flow
Loom : https://www.loom.com/share/84bb44e008ae4a7fa0380088ab662e6c

Figma file : https://www.figma.com/file/fh0OAvj4AS08kHoSxu3DkE/%F0%9F%9A%80-Starknet-Quest?type=design&node-id=3723%3A23765&mode=design&t=T9OoOKnLhybS6HOd-1

### PR DESCRIPTION
**Pull request by Ayoazeez26**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Feature

Resolves: #573

### CODE REVIEW

**Comment by ayushtom** [comment]
@Ayoazeez26 you have a build issue. can you please fix it?

**Comment by Ayoazeez26** [comment]
@ayushtom fixed

**Comment by ayushtom** [comment]
@Ayoazeez26 i think the issue is still there

**Comment by Ayoazeez26** [comment]
@ayushtom  github actions show it deployed successfully and I don't have access to inspect what failed while deploying on vercel

**Comment by Ayoazeez26** [comment]
@ayushtom I believe it's fixed now

File: app/quest/completed/page.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    return item.quests.filter(
+      (quest: { id: number; expired: boolean }) =>
+        !completedQuestIds.includes(quest.id) && !quest.expired
+    );
+  });
+  const incompleteQuests = computeIncompleteQuests.flat();
+
+  const uncompletedQuests = pickRandomObjectsFn({ questArray: incompleteQuests });
+
+  const tweetText =
```
Comment (ayushtom) [inline app/quest/completed/page.tsx]: i believe you will need to encode the text using url encoding to make it work. Right now when you open this link, the text will be obfuscated if you check

File: app/quest/completed/page.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import Lottie from "lottie-react";
+import verifiedLottie from "@public/visuals/verifiedLottie.json";
+import CategoryTitle from "@components/UI/titles/categoryTitle";
+import Quest from "@components/quests/quest";
+
+export default function Page() {
+  const router = useRouter();
+  const { completedQuestIds, categories } = useContext(QuestsContext);
+
+  const computeIncompleteQuests = categories.map((item) => {
```
Comment (ayushtom) [inline app/quest/completed/page.tsx]: can you put this variable in a state ? putting simple variables might cause the variable to recompute again and again on each rerender. I think you can put this in a `useMemo` to avoid this recompute

File: app/quest/completed/page.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    return item.quests.filter(
+      (quest: { id: number; expired: boolean }) =>
+        !completedQuestIds.includes(quest.id) && !quest.expired
+    );
+  });
+  const incompleteQuests = computeIncompleteQuests.flat();
```
Comment (ayushtom) [inline app/quest/completed/page.tsx]: same here. you can use a useMemo

File: app/quest/completed/page.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+  const uncompletedQuests = pickRandomObjectsFn({ questArray: incompleteQuests });
```
Comment (ayushtom) [inline app/quest/completed/page.tsx]: useMemo here too

File: package-lock.json
Comment (ayushtom) [inline package-lock.json:1]: please revert this change if no packages are updated

Comment (Ayoazeez26) [reply package-lock.json:1]: Reverted

File: app/quest/completed/page.tsx (hunk 4 above)
Comment (Ayoazeez26) [reply app/quest/completed/page.tsx]: Done

File: app/quest/completed/page.tsx (hunk 3 above)
Comment (Ayoazeez26) [reply app/quest/completed/page.tsx]: Done

File: app/quest/completed/page.tsx (hunk 2 above)
Comment (Ayoazeez26) [reply app/quest/completed/page.tsx]: Done

File: app/quest/completed/page.tsx (hunk 1 above)
Comment (Ayoazeez26) [reply app/quest/completed/page.tsx]: Updated

**Comment by ayushtom** [comment]
Hey im not sure why the package.json is still in the files changed tab. Undo any changes you are pushing here so it doesn't change the prod deployment behaviour.

**Comment by ayushtom** [comment]
Screenshot 2024-04-04 at 5 18 09 PM
Can you increase the boldness of the "Your NFT is on it's way !"  more? maybe use a bold version of the font?

Also attach a screenshot for when there are no more quests to recommend. It's difficult to recreate that lol

**Comment by Ayoazeez26** [comment]
image

@ayushtom font weight has been increased, and this is a screenshot of when there are no more quests to recommend, also the package-lock file does not show up any longer

**Comment by ayushtom** [comment]
Screenshot 2024-04-05 at 3 21 39 PM
I get this when i have run out of quests to complete. I think this is happening because you aren't checking the length of the array but if the array exists. In this case `[]` might be considered as a valid array and i see an empty screen

**Review by ayushtom — APPROVED** [review]
lgtm
